### PR TITLE
Add OVH provider for external DNS

### PR DESCRIPTION
**Pull request by stephane-monnot**
- Related to rancher/external-dns#105
- Fix rancher/rancher#8735

/!\ :  I have anticipated the release number in the docker-compose.yml, maybe wrong `(image: rancher/external-dns:v0.8.0)`

### CODE REVIEW

File: infra-templates/ovh-dns/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+ovh:
+  image: rancher/external-dns:v0.8.0
```
Comment (rawmind0) [inline infra-templates/ovh-dns/0/docker-compose.yml]: Image released including ovh provider is rancher/external-dns:v0.7.10, could you please update it?? 
```
-  image: rancher/external-dns:v0.8.0
+  image: rancher/external-dns:v0.7.10
```

**Comment by rawmind0** [comment]
Hi @stephane-monnot ... Thanks for the PR and contribute.

Could you please take a look to changes requested??

**Comment by stephane-monnot** [comment]
Done.

File: infra-templates/ovh-dns/0/rancher-compose.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      required: true
+    - variable: "OVH_APPLICATION_KEY"
+      label: "OVH Application Key"
+      description: "Enter your application key"
+      type: "string"
+      required: true
+    - variable: "OVH_APPLICATION_SECRET"
+      label: "OVH Application Secret"
+      description: "Enter your application secret"
+      type: "string"
```
Comment (rawmind0) [inline infra-templates/ovh-dns/0/rancher-compose.yml]: Reviewing the code of ovh provider, `https://github.com/rancher/external-dns/blob/master/providers/ovh/ovh.go` line 43,  `OVH_APPLICATION_SECRET` is required. Could you please add it as required in rancher-compose.yml file?
```
+      required: true
```

**Comment by rawmind0** [comment]
@stephane-monnot before merge PR, could you please set `OVH_APPLICATION_SECRET` as required in rancher-compose.yml file??

**Comment by stephane-monnot** [comment]
Changes has been done.

**Comment by rawmind0** [comment]
Thanks @stephane-monnot ...

LGTM..

**Pull request merged**